vms/platformvm/state/state_test.go

```diff
@@ -687,120 +687,18 @@ func TestPersistStakers(t *testing.T) {
 				test.checkDiffs(require, state, staker, 0 /*height*/)
 
 				// rebuild the state
-				rebuiltState := newStateFromDB(require, db)
-
-				// load relevant quantities
-				require.NoError(rebuiltState.loadCurrentValidators())
-				require.NoError(rebuiltState.loadPendingValidators())
-				require.NoError(rebuiltState.initValidatorSets())
+				rebuiltState := newTestState(t, db)
 
 				// check again that all relevant data are still available in rebuilt state
-				test.checkStakerInState(require, state, staker)
-				test.checkValidatorsSet(require, state, staker)
-				test.checkValidatorUptimes(require, state, staker)
-				test.checkDiffs(require, state, staker, 0 /*height*/)
+				test.checkStakerInState(require, rebuiltState, staker)
+				test.checkValidatorsSet(require, rebuiltState, staker)
+				test.checkValidatorUptimes(require, rebuiltState, staker)
+				test.checkDiffs(require, rebuiltState, staker, 0 /*height*/)
 			})
 		}
 	}
 }
 
-func newInitializedState(require *require.Assertions) State {
```

Comment:
Oddly enough, this wouldn't call `load` previously... so the initialization flow wasn't as expected for a normal initialization.